File: avancado-intermediario/Java/Q4/Universidade.java
public abstract class Universidade{
    private String nome;

    public Universidade(String nome){
        this.nome = nome;
    }

    public String getNome(){
        return this.nome;
    }

    public void setNome(String nome){
        this.nome = nome;
    }

    public abstract void aplicarAvaliacao();
}

File: avancado-intermediario/Python/Q4/universidade.py
from abc import ABC, abstractmethod
class Universidade(ABC):

    def __init__(self, nome):
        self.__nome = nome

    def getNome(self):
        return self.__nome

    def setNome(self, nome):
        self.__nome = nome

    @abstractmethod
    def aplicarAvaliacao(self):
        pass

    def __str__(self) -> str:
        return f"Universidade: {self.getNome()}"